src/Nancy/HttpLink.cs

```diff
@@ -0,0 +1,321 @@
+        /// </summary>
+        public MediaRange Type
+        {
+            get { return this.GetParameterValue<MediaRange>("type"); }
+        }
+
+        /// <summary>
+        /// Indicates whether the current object is equal to another object of the same type.
+        /// </summary>
+        /// <param name="other">An object to compare with this object.</param>
+        /// <returns>
+        /// true if the current object is equal to the <paramref name="other" /> parameter; otherwise, false.
+        /// </returns>
+        public bool Equals(HttpLink other)
+        {
+            if (ReferenceEquals(null, other))
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            if (!Equals(this.targetUri, other.targetUri))
+            {
+                return false;
+            }
+
+            foreach (var parameter in other.Parameters)
+            {
+                object parameterValue;
+                if (!this.parameters.TryGetValue(parameter.Key, out parameterValue))
+                {
+                    return false;
+                }
+
+                if (parameterValue == null || parameter.Value == null)
+                {
+                    continue;
+                }
+
+                if (!parameterValue.Equals(parameter.Value))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Determines whether the specified <see cref="System.Object" />, is equal to this instance.
+        /// </summary>
+        /// <param name="other">The <see cref="System.Object" /> to compare with this instance.</param>
+        /// <returns>
+        ///   <c>true</c> if the specified <see cref="System.Object" /> is equal to this instance; otherwise, <c>false</c>.
+        /// </returns>
+        public override bool Equals(object other)
+        {
+            if (ReferenceEquals(null, other))
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            if (other.GetType() != this.GetType())
+            {
+                return false;
+            }
+
+            return this.Equals((HttpLink)other);
+        }
+
+        /// <summary>
+        /// Returns a hash code for this instance.
+        /// </summary>
+        /// <returns>
+        /// A hash code for this instance, suitable for use in hashing algorithms and data structures like a hash table.
+        /// </returns>
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                var p = 0;
+
+                foreach (var parameter in this.parameters)

```

Comment:
Doesn't `this.parameters` already have an efficient `GetHashCode` implementation?